tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -0,0 +1,160 @@
+    internal static void Analyze(IHeaderDictionary responseHeaders, IntegrationId integrationId, string serviceName, int responseCode)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(serviceName) || IsIgnorableResponseCode((HttpStatusCode)responseCode))
+            {
+                return;
+            }
+
+            IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
+
+            string contentTypeValue = string.Empty;
+            string contentOptionValue = string.Empty;
+
+            // We iterate instead of trying to get the key directly because keys are case insensitive
+            foreach (var header in responseHeaders)
+            {
+                var headerKeyLow = header.Key.ToLowerInvariant();
+                if (headerKeyLow == ContentTypeLow)
+                {
+                    contentTypeValue = header.Value;
+
+                    if (!IsHtmlResponse(contentTypeValue))
+                    {
+                        return;
+                    }
+                }
+
+                if (headerKeyLow == XContentTypeOptionsLow)
+                {
+                    contentOptionValue = header.Value;
+
+                    if (IsNoSniffContentOptions(contentOptionValue))
+                    {
+                        return;
+                    }
+                }
+            }
+
+            LaunchXContentTypeOptionsVulnerability(integrationId, serviceName, contentTypeValue, contentOptionValue);
+        }
+        catch (Exception error)
+        {
+            Log.Error(error, $"{nameof(ReturnedHeadersAnalyzer)}.{nameof(Analyze)} exception");
+        }
+    }
+#endif
+
+    private static void LaunchXContentTypeOptionsVulnerability(IntegrationId integrationId, string serviceName, string contentTypeValue, string contentOptionValue)
+    {
+        if (!string.IsNullOrEmpty(contentTypeValue))
+        {
+            IastModule.OnXContentTypeOptionsHeaderMissing(integrationId, contentOptionValue, serviceName);
+        }
+    }
+
+    private static bool IsHtmlResponse(string contentTypeValue)
+    {
+        var contentType = contentTypeValue.ToLowerInvariant();
+        return contentType.Contains("text/html") || contentType.Contains("application/xhtml+xml");
+    }
+
+    private static bool IsNoSniffContentOptions(string contentOptionValue)
+    {
+        return contentOptionValue.ToLowerInvariant() == "nosniff";
+    }
+
+    public static bool IsIgnorableResponseCode(HttpStatusCode httpStatus)
+    {
+        switch (httpStatus)
+        {
+            case HttpStatusCode.Moved:
+            case HttpStatusCode.Redirect:
+            case HttpStatusCode.NotModified:
+            case HttpStatusCode.TemporaryRedirect:
+            case HttpStatusCode.NotFound:
+            case HttpStatusCode.Gone:
+            case HttpStatusCode.InternalServerError:
```

Comment:
They have _multiple_ `HttpStatusCode`s with the same value. I _assume_ we don't need both, but I'm not 100%. Maybe this is safer? 
```suggestion
            case HttpStatusCode.Moved:
            case HttpStatusCode.MovedPermanently:
            case HttpStatusCode.Found:
            case HttpStatusCode.Redirect:
            case HttpStatusCode.NotModified:
            case HttpStatusCode.TemporaryRedirect:
            case HttpStatusCode.RedirectKeepVerb:
            case HttpStatusCode.NotFound:
            case HttpStatusCode.Gone:
            case HttpStatusCode.InternalServerError:
```

Reply:
It's interesting: the compiler complains about this (The switch statement contains multiple cases with the label value '301')

Reply:
Good to know, in which case ignore this, thanks for checking! 😄